scripts/install.sh

```diff
@@ -5,64 +5,83 @@
 
 set -euo pipefail
 
-# Get the latest version of the app from the GitHub API
+# --- Get the latest version of the app from the GitHub API
+
 get_latest_version() {
-    curl -s https://api.github.com/repos/amir1376/ab-download-manager/releases/latest | jq -r '.tag_name'
+    curl -fSs https://api.github.com/repos/amir1376/ab-download-manager/releases/latest | jq -r '.tag_name'
 }
 
 APP_NAME="ABDownloadManager"
 PLATFORM="linux"
 EXT="tar.gz"
+
 LATEST_VERSION=$(get_latest_version)
 ASSET_NAME=${APP_NAME}_${LATEST_VERSION:1}_${PLATFORM}.${EXT}
 BINARY_PATH=$HOME/.local/$APP_NAME/bin/$APP_NAME
 
-# Check if curl, jq, and tar are installed
-check_dependencies() {
-    for cmd in curl jq tar; do
-        if ! command -v "$cmd" &>/dev/null; then
-            print_message "Error: $cmd is not installed. Please install it and try again."
-            exit 1
-        fi
-    done
-}
+DOWNLOAD_URL="https://github.com/amir1376/ab-download-manager/releases/download/${LATEST_VERSION}/$ASSET_NAME"
+DEPENDENCIES=(curl jq tar)
+
+LOG_FILE="/tmp/ab-dm-installer.log"
+
+# --- Custom Logger
+logger() {
+  timestamp=$(date +"%Y/%m/%d %H:%M:%S")
 
-# Get the download URL for the latest version of the app
-get_download_url() {
-    echo "https://github.com/amir1376/ab-download-manager/releases/download/${LATEST_VERSION}/$ASSET_NAME"
+  if [[ "$1" == "error" ]]; then
+    # Red color for errors
+    echo -e "${timestamp} -- "$0" [Error]: \033[0;31m$@\033[0m" | tee -a ${LOG_FILE}
+  else
+    # Default color for non-error messages
+    echo -e "${timestamp} -- "$0" [Info]: $@" | tee -a ${LOG_FILE}
+  fi
 }
 
-close_if_running() {
-    pid=$(pidof -s "$APP_NAME")
+# --- Detect OS and The Package Manager to use
+detect_package_manager() {
+    if [ -f /etc/os-release ]; then
+        source /etc/os-release
+        local OS=${NAME}
+    elif type lsb_release >/dev/null 2>&1; then
+        local OS=$(lsb_release -si)
+    elif [ -f /etc/lsb-release ]; then
+        source /etc/lsb-release
+        local OS="${DISTRIB_ID}"
+    elif [ -f /etc/debian_version ]; then
+        local OS=Debian
+    else
+        logger error "Your Linux Distro is not Supperted."
+        logger error "Please install ${DEPENDENCIES[@]} Manually."
+        exit 1
+    fi
 
-    if [ -n "$pid" ]; then
-        kill -9 "$pid"
-        echo "Closed running instance of AB Download Manager before updating."
+    if `grep -E 'Debian|Ubuntu' <<< $OS > /dev/null` ; then
+        systemPackage="apt"
+    elif `grep -E 'Fedora|CentOS|Red Hat|AlmaLinux' <<< $OS > /dev/null`; then
+        systemPackage="dnf"
     fi
 }
 
-# Delete the old version of the app if it exists
+# --- Install dependencies
+install_dependencies() {
+    detect_package_manager
+    logger "installing dependencies: ${DEPENDENCIES[@]}"
+    sudo ${systemPackage} install -y ${DEPENDENCIES[@]}
+}
```

Comment:
Please make it optional since it requires `sudo`. its better to check for the dependencies first.
if they are missing. then you can ask the user to press "Y" to install them automatically (with the following command) or "N" to cancel the installation

Reply:
the reason I didn't do that is : 

- the 3 dependencies which are `curl , jq, tar` these 3 packages are already installed in most Linux distributions. so here there is no need to ask
- the package manager checks the presence of packages 
- if there is a non-popular package the script intended to install, the script has to wait for the user's response.